src/events.c

```diff
@@ -0,0 +1,103 @@
+#include "events.h"
+
+/* this is only to queue GDK key events, in order to later send them if the map didn't match */
+static struct {
+    GdkEventKey **queue;                /* queue holding submitted events */
+    int            qlen;                /* pointer to last char in queue */
+    bool           processing;          /* whether or not events are processing */
+} events;
+
+/**
+ * Append an event into the queue.
+ */
+void queue_event(GdkEventKey* e)
+{
+    if (vb.mode->id != 'i') {
```

Comment:
The reason I'm casing out input mode here is because command mode (and possibly other modes) seem to be respecting incorrect maps fine. The initial goal of this PR is just to fix #258. Eventually this will be available for all input modes.